### ISSUE TITLE
fetchNextPage from useInfiniteQuery doesn't work if has been fired before the first query returned data

### ISSUE THREAD

**Issue by furagi** [TanStack/query#5583]
### Describe the bug

`fetchNextPage` from `useInfiniteQuery` doesn't send a request if was fired before the first request returned data.

If you open the codesandbox example, you will see that query results looks like we had just one call. However, if you check line 61 (or check logs in console) you will see that there were two requests - first ran with timeout 0ms, the second one - with timeout 1ms. Also, if you check line 30, you can see, that every request takes 100ms to resolve.
Now, if on line 61 you change the delay multiplier from 1 to 1000 (so `fetchNextPage` will be called when we already have the first page), you will see that both requests were fired (as expected).

Unfortunately, I wasn't able to attach screenshots, because github throws an error when I'm trying to do so.

### Your minimal, reproducible example

https://codesandbox.io/p/sandbox/snowy-feather-xzqyy9?embed=1&file=%2Fsrc%2Findex.jsx%3A1%2C1

### Steps to reproduce

1. Open the codesandbox example

### Expected behavior

As a user, I expect that if I call `fetchNextPage` while the very first request is still loading, the second request will be sent as well.

### How often does this bug happen?

Every time

### Platform

- OS: MacOS
- Browser: Google Chrome
- Version 114.0.5735.106 (Official Build) (arm64)

### Tanstack Query adapter

react-query

### TanStack Query version

4.19.0

### TypeScript version

4.9.5

### Additional context

The problem I'm trying to solve is that I need to fetch a dynamic amount of pages on demand (and also every page has specific params - so it's impossible to calculate the next page params from the existing ones). It works perfect if I fetch page by page, but sometimes I need to fetch few pages as the app loaded (so I do something like `pagesArray.forEach(fetchPage)`, where `fetchPage` is a function that sets first page params if we didn't fetch anything at all - otherwise it calls `fetchNextPage`). And once I do this, only the first request is fired.

**Comment by TkDodo** [comment]
I don't fully understand the use-case, but here are some constraints:

- there can always only be one fetch going on for an InfiniteQuery. There is only one cache entry, split up into multiple pages. You can't fetch two pages at the same time, they would likely overwrite each other.
- That's why `fetchNextPage` has an option `cancelRefetch`, that defaults to `true`. When you call `fetchNextPage`, it will cancel a currently running request for that query so that it can append the next page at the end.
- If you call `fetchNextPage` in a loop without awaiting each fetch, it will not work. They have to run in serial. A `setTimeout` is not good enough.
- You can set `cancelRefetch: false` as a param to `fetchNextPage`, but then it will just pick up any running promises and not cancel the ongoing one.

So, basically, this:

> As a user, I expect that if I call fetchNextPage while the very first request is still loading, the second request will be sent as well.

is not possible.

**Comment by TkDodo** [comment]
I see. Would you like to make a PR to the docs?

**Comment by TkDodo** [comment]
if you want to improve the docs, please go ahead.
